src/main/java/graphql/schema/AbstractGraphQLType.java

```diff
@@ -0,0 +1,32 @@
+package graphql.schema;
+
+import static graphql.Assert.assertNotNull;
+
+public class AbstractGraphQLType implements GraphQLType {
```

Comment:
I favor this introduction of a common abstract type to keep that logic in a single spot and to avoid lots of code duplication.